tests/test_router_bgp.rb

```diff
@@ -399,12 +400,45 @@ def test_default_cluster_id
   def test_enforce_first_as
     asnum = 55
     bgp = RouterBgp.new(asnum)
-    bgp.enforce_first_as = true
-    assert(bgp.enforce_first_as,
-           'bgp enforce-first-as should be enabled')
-    bgp.enforce_first_as = false
-    refute(bgp.enforce_first_as,
-           'bgp enforce-first-as should be disabled')
+    if platform == :ios_xr
+      # Lets assert based on XR standard output if a command is not found
+      pattern = 'No such configuration item'
+
+      # Test default running-config test
+      s = @device.cmd("show running-config router bgp #{asnum} " \
+                      'bgp enforce-first-as disable')
+      assert_match(pattern, s,
+                   'bgp enforce-first-as disable should not be enabled')
+
+      # Disable enforce-first-as running-config test
+      bgp.enforce_first_as = false
+      s = @device.cmd("show running-config router bgp #{asnum} " \
+                      'bgp enforce-first-as disable')
+      refute_match(pattern, s,
+                   'bgp enforce-first-as disable should be enabled')
+
+      # Enable enforce-first-as running-config test
+      bgp.enforce_first_as = true
+      s = @device.cmd("show running-config router bgp #{asnum} " \
+                      'bgp enforce-first-as disable')
+      assert_match(pattern, s,
+                   'bgp enforce-first-as disable should not be enabled')
+
+      # Test node utils reporting
+      bgp.enforce_first_as = true
+      assert(bgp.enforce_first_as,
+             'bgp enforce-first-as disable should not be enabled')
+      bgp.enforce_first_as = false
+      refute(bgp.enforce_first_as,
+             'bgp enforce-first-as disable should be enabled')
+    else
+      bgp.enforce_first_as = true
+      assert(bgp.enforce_first_as,
+             'bgp enforce-first-as should be enabled')
+      bgp.enforce_first_as = false
+      refute(bgp.enforce_first_as,
+             'bgp enforce-first-as should be disabled')
```

Comment:
This still seems a lot clunkier than it needs to be. If you're wanting to test both CLI and node_util class, you should test both for both OS. The 'cmd' and 'pattern' will be different for XR vs NXOS but all of the other test logic should be the same... something like:

```ruby
if platform == :ios_xr
  @default_show_command = "show running-config router bgp #{asnum} " \
                      'bgp enforce-first-as disable'
   @default_output_pattern = 'bgp enforce-first-as disable'
elsif platform == :nexus
  @default_show_cmd = "show run bgp all"
  @default_output_pattern = "...."
end
# COMMON CODE
bgp.enforce_first_as = true
assert(bgp.enforce_first_as)
refute_show_match(msg: 'bgp enforce-first-as should be enabled')

bgp.enforce_first_as = false
refute(bgp.enforce_first_as)
assert_show_match(msg: 'bgp enforce-first-as should be disabled')
```

(ref: https://github.com/cisco/cisco-network-node-utils/blob/develop/docs/README-develop-best-practices.md#mt5-make-use-of-the-assert_show_match-and-refute_show_match-helper-methods-to-validate-expected-outcomes-in-the-cli-instead-of-devicecmdshow)